wled00/pin_manager.h

```diff
@@ -4,26 +4,103 @@
  * Registers pins so there is no attempt for two interfaces to use the same pin
  */
 #include <Arduino.h>
+#include "const.h" // for USERMOD_* values
+
+typedef struct PinManagerPinType {
+  int8_t  pin;
+  uint8_t isOutput;
+} managed_pin_type;
+
+/*
+ * Allows PinManager to "lock" an allocation to a specific
+ * owner, so someone else doesn't accidentally de-allocate
+ * a pin it hasn't allocated.  Also enhances debugging.
+ * 
+ * RAM Cost:
+ *     17 bytes on ESP8266
+ *     40 bytes on ESP32
+ */
+enum struct PinOwner : uint8_t {
```

Comment:
Reduced (vs. prior PR) RAM overhead by changing to single byte.

Built-in owners have high bit set to one (0x80).
User modules use their assigned user module ID.